src/graphics/aurora/model_kotor.cpp

```diff
@@ -472,10 +506,64 @@ void ModelNode_KotOR::load(Model_KotOR::ParserContext &ctx) {
 	}
 }
 
-void ModelNode_KotOR::readNodeControllers(Model_KotOR::ParserContext &UNUSED(ctx),
-		uint32 UNUSED(offset), uint32 UNUSED(count), std::vector<float> &UNUSED(data)) {
-
-	// TODO
+void ModelNode_KotOR::readNodeControllers(Model_KotOR::ParserContext &ctx,
+		uint32 offset, uint32 count, std::vector<float> &data) {
+	uint32 pos = ctx.mdl->seek(offset);
+	for (uint32 i = 0; i < count; i++) {
+		uint32 type = ctx.mdl->readUint32LE();
+		ctx.mdl->skip(2);
+		uint16 rowCount = ctx.mdl->readUint16LE();
+		uint16 timeIndex = ctx.mdl->readUint16LE();
+		uint16 dataIndex = ctx.mdl->readUint16LE();
+		uint8 columnCount = ctx.mdl->readByte();
+		ctx.mdl->skip(3);
+		switch (type) {
+			case kControllerTypePosition:
+				switch (columnCount) {
```

Comment:
I think the inner switches should move to their own methods. Too much intentation here to make code easily readable.

Reply:
Fixed